File: firefly-example/src/main/java/com/firefly/example/reactive/coffee/store/router/RouterInstaller.java
package com.firefly.example.reactive.coffee.store.router;

/**
 * @author Pengtao Qiu
 */
public interface RouterInstaller extends Comparable<RouterInstaller> {

    void install();

    default int compareTo(RouterInstaller o) {
        return order().compareTo(o.order());
    }

    default Integer order() {
        return Integer.MAX_VALUE;
    }
}

File: firefly-example/src/main/kotlin/com/firefly/example/kotlin/coffee/store/router/RouterInstaller.kt
package com.firefly.example.kotlin.coffee.store.router

/**
 * @author Pengtao Qiu
 */
interface RouterInstaller : Comparable<RouterInstaller> {

    fun install()

    fun order() = Integer.MAX_VALUE

    override fun compareTo(other: RouterInstaller): Int = order().compareTo(other.order())
}